pkg/api/defaults_test.go

```diff
@@ -914,26 +914,125 @@ func TestSetVMSSDefaultsAndZones(t *testing.T) {
 }
 
 func TestAKSDockerEngineDistro(t *testing.T) {
```

Comment:
@CecileRobertMichon start by reading these tests to get a sense of what this PR delivers